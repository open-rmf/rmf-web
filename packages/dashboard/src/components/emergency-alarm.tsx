import { IconButton } from '@material-ui/core';
import React from 'react';
import NotificationsActiveIcon from '@material-ui/icons/NotificationsActive';

export interface EmergencyAlarmProps {
  isActive: boolean | null;
}

/**
 * TODO: This component represents the emergency alarm. It is kept as a separate component in case
 * more logic needs to be added in the future, such as turning the alarm on and off. This component
 * is on hold until an appropriate backend is available to manage the alarm.
 *
 */
export const EmergencyAlarm = React.memo((props: EmergencyAlarmProps) => {
  const { isActive } = props;
  return (
    <>
      {isActive && (
        <IconButton id="alarm-btn" color={'secondary'}>
          <NotificationsActiveIcon />
        </IconButton>
      )}
    </>
  );
});
