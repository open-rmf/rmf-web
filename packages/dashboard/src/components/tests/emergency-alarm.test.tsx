import { IconButton } from '@material-ui/core';
import { mount, shallow } from 'enzyme';
import React from 'react';
import { EmergencyAlarm } from '../emergency-alarm';

describe('Emergency alarm behavior', () => {
  let root: any;

  afterEach(() => {
    root.unmount();
  });

  test('Alarm renders correctly when is activated', () => {
    root = shallow(<EmergencyAlarm isActive={true}></EmergencyAlarm>);
    expect(root).toMatchSnapshot();
  });

  test('Alarm icon is not displayed when is no activated', () => {
    root = mount(<EmergencyAlarm isActive={null}></EmergencyAlarm>);
    expect(root.find(IconButton).exists()).toBe(false);
  });

  test('Alarm icon displays a red icon when is activated', () => {
    root = mount(<EmergencyAlarm isActive={true}></EmergencyAlarm>);
    expect(root.find(IconButton).prop('color')).toBe('secondary');
  });
});
