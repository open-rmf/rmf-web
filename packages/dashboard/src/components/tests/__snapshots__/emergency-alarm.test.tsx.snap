// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emergency alarm behavior Alarm renders correctly when is activated 1`] = `
<Fragment>
  <WithStyles(ForwardRef(IconButton))
    color="secondary"
    id="alarm-btn"
  >
    <NotificationsActiveIcon />
  </WithStyles(ForwardRef(IconButton))>
</Fragment>
`;
